packages/directory/addon/controllers/directory.ts

```diff
@@ -58,42 +51,56 @@ export default class DirectoryController extends Controller {
    */
   q = '';
 
+  /**
+   * @property {String} sortKey - sort key (computed by sortBy query param)
+   */
+  get sortKey() {
+    const { sortBy } = this;
+    return sortBy === 'author' ? 'author.id' : sortBy;
+  }
+
   /**
    * @property {String} title - Title for the table
    */
   @computed('filter', 'router.currentRouteName')
   get title() {
-    const currentRoute = get(this, 'router.currentRouteName'),
-      dirInfo = this.directories.getDirectories(),
-      currentDir = arr(dirInfo).findBy('routeLink', currentRoute);
+    const { router, directories, filter } = this;
+    const currentDir = directories.getDirectories().find(dir => dir.routeLink === router.currentRouteName);
 
-    let title = currentDir.name,
-      queryParams = getProperties(this, ['filter']),
-      match = currentDir.filters.filter(filter => JSON.stringify(filter.queryParam) === JSON.stringify(queryParams));
+    const queryParams = { filter };
+    const match = currentDir?.filters?.filter(
+      filter => JSON.stringify(filter.queryParams) === JSON.stringify(queryParams)
+    );
 
-    if (match.length === 1) {
-      title = match[0].name;
+    if (match?.length === 1) {
+      return match[0].name;
     }
 
-    return title;
+    return currentDir?.name;
   }
 
   /**
    * @action searchFor
    * Sets the query param for search
-   * @param {String} query
+   * @param {string} query
    */
   @action
-  searchFor(query) {
+  searchFor(query: string) {
     set(this, 'q', query);
   }
 
   /**
    * @action updateQueryParams - update to the new query params
-   * @param {Object} queryParams
+   * @param {object} queryParams
    */
   @action
-  updateQueryParams(queryParams) {
+  updateQueryParams(queryParams: object) {
```

Comment:
`{ [key: string]: string; }`

Reply:
I think query params can actually be `boolean | number | array | string`

[ember queryparams guides](https://guides.emberjs.com/release/routing/query-params/)
[current type definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/80fef0454291a5c8cd70a4fcaae767b4da60314f/types/ember__controller/index.d.ts#L13)

But we could add a type for DirectoryQueryParams explicitly

```ts
type DirectoryQueryParams = {
  filter: string | undefined,
  type: string | undefined,
  sortBy: string,
  sortDir: string,
  q: string
}
```